catalog/dags/data_refresh/copy_data.py

```diff
@@ -0,0 +1,321 @@
+    temp_table_name: str,
+    deleted_table_name: str,
+):
+    """
+    Copy an individual table from the upstream DB into a new temporary table
+    in the downstream DB.
+    """
+    shared_cols = get_shared_columns(
+        upstream_conn_id=upstream_conn_id,
+        downstream_conn_id=downstream_conn_id,
+        upstream_table_name=upstream_table_name,
+        downstream_table_name=downstream_table_name,
+    )
+
+    schema = create_schema(
+        downstream_conn_id=downstream_conn_id,
+        upstream_table_name=upstream_table_name,
+    )
+
+    create_temp_table = _run_sql.override(task_id="create_temp_table")(
+        postgres_conn_id=downstream_conn_id,
+        sql_template=queries.CREATE_TEMP_TABLE_QUERY,
+        temp_table_name=temp_table_name,
+        downstream_table_name=downstream_table_name,
+    )
+
+    setup_id_columns = _run_sql.override(task_id="setup_id_columns")(
+        postgres_conn_id=downstream_conn_id,
+        sql_template=queries.ID_COLUMN_SETUP_QUERY,
+        temp_table_name=temp_table_name,
+    )
+
+    setup_tertiary_columns = _run_sql.override(task_id="setup_tertiary_columns")(
+        postgres_conn_id=downstream_conn_id,
+        sql_template=tertiary_column_query,
+        temp_table_name=temp_table_name,
+    )
+
+    copy = copy_data.override(execution_timeout=timeout)(
+        postgres_conn_id=downstream_conn_id,
+        limit=limit,
+        sql_template=copy_data_query,
+        temp_table_name=temp_table_name,
+        schema_name=schema,
+        upstream_table_name=upstream_table_name,
+        deleted_table_name=deleted_table_name,
+        columns=shared_cols,
+    )
+
+    add_primary_key = _run_sql.override(task_id="add_primary_key")(
+        postgres_conn_id=downstream_conn_id,
+        sql_template=queries.ADD_PRIMARY_KEY_QUERY,
+        temp_table_name=temp_table_name,
+    )
+
+    create_temp_table >> setup_id_columns >> setup_tertiary_columns
+    setup_tertiary_columns >> copy
+    copy >> add_primary_key
+    return
+
+
+@task_group(group_id="copy_upstream_tables")
+def copy_upstream_tables(
+    environment: Environment, data_refresh_config: DataRefreshConfig
+):
+    """
+    For each upstream table associated with the given media type, create a new
+    temp table in the downstream DB and copy all the upstream data into it.
+    These temp tables will later replace the main media tables in the API.
+
+    This task does _not_ apply all indices and constraints, merely copies
+    the data.
+    """
+    downstream_conn_id = POSTGRES_API_CONN_IDS.get(environment)
+    upstream_conn_id = POSTGRES_CONN_ID
+
+    create_fdw = create_fdw_extension(downstream_conn_id=downstream_conn_id)
+
+    init_fdw = initialize_fdw(
+        upstream_conn_id=upstream_conn_id,
+        downstream_conn_id=downstream_conn_id,
+    )
+
+    limit = get_record_limit()
+
+    # Copy all tables mapped for this media type
+    copy_tables = copy_upstream_table.partial(
```

Comment:
Is it possible to add the table name to the task group? Or some way to make it easier to identify the difference in the DAG flow? When I look at the following screen, for example, I don't see anything obvious making it clear what the important/"key" variable is between the mapped tasks:

![image](https://github.com/WordPress/openverse/assets/24264157/37aa784a-6bc2-47b5-b0aa-5b35111dfb68)

Maybe there's a different way to do what I am asking than using the task group name, I don't want to necessarily expand out the tasks into separately named tasks per table... but it's hard to tell at a glance (basically without looking at logs, as far as I can tell) what each mapped task is for.

Reply:
This has bugged me in several of our DAGs in the past! I was going to tell you this will be available in an upcoming version of Airflow, but it turns out it's in 2.9 so we've recently updated and should have that ability :) I'm having some trouble getting it to work as documented but will keep playing around with it.

Reply:
Got it! I'll have to go fix this in our other DAGs now that the feature is available :)